Mathlib/AlgebraicGeometry/Morphisms/Separated.lean

```diff
@@ -86,7 +85,7 @@ instance (R S : CommRingCat.{u}) (f : R ⟶ S) : IsSeparated (Spec.map f) := by
     (Algebra.TensorProduct.lmul'_apply_tmul (R := R) (S := S) 1 x).trans (one_mul x)⟩
 
 instance (priority := 100) [h : IsAffineHom f] : IsSeparated f := by
-  clear g Z
+  clear * - h
```

Comment:
It's sad that these `wlog` proofs break every time you add a section variable.

Reply:
I think this change is immune to future alterations.

Reply:
I am still not a big fan of `clear` instead of restructuring section variables, although I thought that the new variable inclusion mechanism should avoid this behavior? (`g`, `Z`, `W` are not mentioned in the statement).

Reply:
I think in this case it amounts to moving the composition instance from line 65 near `IsSeparated.of_comp` and make a section there with `variable {Z} (g : Y ⟶ Z)`. Then you have to add one `{Z} (g : Y ⟶ Z)` in L102 and one `{Z W}` in L149.

Reply:
How does the new commit look?
(The new variable inclusion mechanism only works for `theorem` and not `instance`)

Reply:
Oh I did not notice that. This is a good solution then.